tests/components/mqtt/test_init.py

```diff
@@ -64,619 +58,642 @@ def mock_mqtt():
+async def test_subscribe_topic_sys_root(hass, mqtt_mock, calls, record_calls):
+    """Test the subscription of $ root topics."""
+    await mqtt.async_subscribe(hass, "$test-topic/subtree/on", record_calls)
 
-        self.hass.block_till_done()
-        assert len(self.calls) == 0
+    async_fire_mqtt_message(hass, "$test-topic/subtree/on", "test-payload")
 
-    def test_subscribe_topic_sys_root(self):
-        """Test the subscription of $ root topics."""
-        mqtt.subscribe(self.hass, "$test-topic/subtree/on", self.record_calls)
+    await hass.async_block_till_done()
+    assert len(calls) == 1
+    assert calls[0][0].topic == "$test-topic/subtree/on"
+    assert calls[0][0].payload == "test-payload"
 
-        fire_mqtt_message(self.hass, "$test-topic/subtree/on", "test-payload")
 
-        self.hass.block_till_done()
-        assert len(self.calls) == 1
-        assert self.calls[0][0].topic == "$test-topic/subtree/on"
-        assert self.calls[0][0].payload == "test-payload"
+async def test_subscribe_topic_sys_root_and_wildcard_topic(
+    hass, mqtt_mock, calls, record_calls
+):
+    """Test the subscription of $ root and wildcard topics."""
+    await mqtt.async_subscribe(hass, "$test-topic/#", record_calls)
 
-    def test_subscribe_topic_sys_root_and_wildcard_topic(self):
-        """Test the subscription of $ root and wildcard topics."""
-        mqtt.subscribe(self.hass, "$test-topic/#", self.record_calls)
+    async_fire_mqtt_message(hass, "$test-topic/some-topic", "test-payload")
 
-        fire_mqtt_message(self.hass, "$test-topic/some-topic", "test-payload")
+    await hass.async_block_till_done()
+    assert len(calls) == 1
+    assert calls[0][0].topic == "$test-topic/some-topic"
+    assert calls[0][0].payload == "test-payload"
 
-        self.hass.block_till_done()
-        assert len(self.calls) == 1
-        assert self.calls[0][0].topic == "$test-topic/some-topic"
-        assert self.calls[0][0].payload == "test-payload"
 
-    def test_subscribe_topic_sys_root_and_wildcard_subtree_topic(self):
-        """Test the subscription of $ root and wildcard subtree topics."""
-        mqtt.subscribe(self.hass, "$test-topic/subtree/#", self.record_calls)
+async def test_subscribe_topic_sys_root_and_wildcard_subtree_topic(
+    hass, mqtt_mock, calls, record_calls
+):
+    """Test the subscription of $ root and wildcard subtree topics."""
+    await mqtt.async_subscribe(hass, "$test-topic/subtree/#", record_calls)
 
-        fire_mqtt_message(self.hass, "$test-topic/subtree/some-topic", "test-payload")
+    async_fire_mqtt_message(hass, "$test-topic/subtree/some-topic", "test-payload")
 
-        self.hass.block_till_done()
-        assert len(self.calls) == 1
-        assert self.calls[0][0].topic == "$test-topic/subtree/some-topic"
-        assert self.calls[0][0].payload == "test-payload"
+    await hass.async_block_till_done()
+    assert len(calls) == 1
+    assert calls[0][0].topic == "$test-topic/subtree/some-topic"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)